src/guardian/block-guard/block-guard.service.ts

```diff
@@ -68,14 +82,42 @@ export class BlockGuardService {
     const endTimer = this.blockRequestsHistogram.startTimer();
     try {
       const guardianAddress = this.securityService.getGuardianAddress();
+      const [
+        depositRoot,
+        depositedEvents,
+        guardianIndex,
+        lidoWC,
+        securityVersion,
+      ] = await Promise.all([
+        this.depositService.getDepositRoot({ blockHash }),
+        this.depositService.getAllDepositedEvents(blockNumber, blockHash),
+        this.securityService.getGuardianIndex({ blockHash }),
+        this.lidoService.getWithdrawalCredentials({ blockHash }),
+        this.securityService.version({
+          blockHash,
+        }),
+      ]);
 
-      const [depositRoot, depositedEvents, guardianIndex, lidoWC] =
-        await Promise.all([
-          this.depositService.getDepositRoot({ blockHash }),
-          this.depositService.getAllDepositedEvents(blockNumber, blockHash),
-          this.securityService.getGuardianIndex({ blockHash }),
-          this.lidoService.getWithdrawalCredentials({ blockHash }),
-        ]);
+      const theftHappened =
+        await this.stakingModuleGuardService.getHistoricalFrontRun(
+          depositedEvents,
+          lidoWC,
+        );
+
+      const alreadyPausedDeposits = await this.alreadyPausedDeposits(
+        blockHash,
+        securityVersion,
+      );
```

Comment:
I don't agree that we need to make request optimistically
If reason is not absence of version method, it will not work properly 
we will not get a lot of profit from it